lib/metasploit/framework/data_service/remote/http/core.rb

```diff
@@ -239,7 +242,10 @@ def build_client_pool(size)
     @client_pool = Queue.new()
     (1..size).each {
       http = Net::HTTP.new(@endpoint.host, @endpoint.port)
-      http.use_ssl = true if @endpoint.use_ssl
+      if @endpoint.is_a?(URI::HTTPS)
+        http.use_ssl = true
+        http.verify_mode = OpenSSL::SSL::VERIFY_NONE
```

Comment:
Should `http.verify_mode` be set `OpenSSL::SSL::VERIFY_PEER`? Without that no verification will occur, correct? If the user provides a cert and key we need to assign those to `http.cert` and `http.key`; otherwise, assume they are using a cert that can be verified through the CA certificates installed on their system by default. Another option would be to support custom CA certs via `http.cert_store` and another parameter.
See [net-http-cheat-sheet/ssl_and_https](https://github.com/augustl/net-http-cheat-sheet/blob/master/ssl_and_https.rb)

Reply:
Yeah, I wasn't really sure how to get around this. Obviously for my testing I only had a self-signed cert, so this option was necessary. One thing I was thinking about doing was enabling cert verification by defaults adding a flag to the command to override it. What do you think about that?

In the meantime I'll check into the things you suggested. I wasn't aware of all of them.

Reply:
Wouldn't verification of self-signed certificates work without issue if you provided the cert and key to both the server and client sides?